src/types/ErrorRate/ErrorRate.ts

```diff
@@ -3,9 +3,16 @@ import { RateHealthConfig, ToleranceConfig } from '../ServerConfig';
 import { checkExpr, emptyRate, getRateHealthConfig, getErrorCodeRate } from './utils';
 import { ErrorRatio, Rate, RequestTolerance } from './types';
 
-// Sum the inbound and outbound request for calculate the global status
+// Sum the inbound and outbound request for calculating the global status
 export const sumRequests = (inbound: RequestType, outbound: RequestType): RequestType => {
-  let result: RequestType = inbound;
+  let result: RequestType = {};
+  // init result with a deep clone of inbound
+  for (let [protocol, req] of Object.entries(inbound)) {
+    result[protocol] = {};
+    for (let [code, rate] of Object.entries(req)) {
+      result[protocol][code] = rate;
+    }
+  }
```

Comment:
Good one, I didn't realize this :(

Reply:
@aljesusg This was a tricky one to figure out.  But yes, the by-reference assignment led to corruption of the inbound rates.